sccache: error: thread 'tokio-runtime-worker' panicked at src/lru_disk_cache/mod.rs:204:17: Error removing file from cache:

In [this build](https://github.com/hylo-lang/llvm-build/actions/runs/7895297454/job/21547584509#step:14:2546), sccache freaked out as follows, only on macos (full logs:
[5_distribution (macos-latest, amd64).txt](https://github.com/mozilla/sccache/files/14275159/5_distribution.macos-latest.amd64.txt)
[6_distribution (macos-latest, arm64).txt](https://github.com/mozilla/sccache/files/14275160/6_distribution.macos-latest.arm64.txt)

You can reproduce by forking that project at 90b9e81 and manually triggering the workflow
```
sccache: encountered fatal error
sccache: error: thread 'tokio-runtime-worker' panicked at src/lru_disk_cache/mod.rs:204:17: Error removing file from cache: `"/Users/runner/work/llvm-build/llvm-build/.sccache/preprocessor/7/d/5/7d553347d23a053780beb2e6cc81746db44ea0eee6b9ee645fc2bb29a43d8c4c"`: failed to remove file `/Users/runner/work/llvm-build/llvm-build/.sccache/preprocessor/7/d/5/7d553347d23a053780beb2e6cc81746db44ea0eee6b9ee645fc2bb29a43d8c4c`
sccache: caused by: thread 'tokio-runtime-worker' panicked at src/lru_disk_cache/mod.rs:204:17: Error removing file from cache: `"/Users/runner/work/llvm-build/llvm-build/.sccache/preprocessor/7/d/5/7d553347d23a053780beb2e6cc81746db44ea0eee6b9ee645fc2bb29a43d8c4c"`: failed to remove file `/Users/runner/work/llvm-build/llvm-build/.sccache/preprocessor/7/d/5/7d553347d23a053780beb2e6cc81746db44ea0eee6b9ee645fc2bb29a43d8c4c`
...
sccache: encountered fatal error
sccache: error: thread 'tokio-runtime-worker' panicked at src/cache/disk.rs:180:14: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
sccache: caused by: thread 'tokio-runtime-worker' panicked at src/cache/disk.rs:180:14: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
...
ache: encountered fatal error
sccache: error: thread 'tokio-runtime-worker' panicked at src/cache/disk.rs:180:14: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
sccache: caused by: thread 'tokio-runtime-worker' panicked at src/cache/disk.rs:180:14: called `Result::unwrap()` on an `Err` value: PoisonError { .. }
```

--- a/src/lru_disk_cache/mod.rs
+++ b/src/lru_disk_cache/mod.rs
@@ -225,7 +225,22 @@ impl LruDiskCache {
             //TODO: check that files are removable during `init`, so that this is only
             // due to outside interference.
             fs::remove_file(&remove_path).unwrap_or_else(|e| {
-                panic!("Error removing file from cache: `{:?}`: {}", remove_path, e)
+                // Sometimes the file has already been removed
+                // this seems to happen when the max cache size has been reached
+                // https://github.com/mozilla/sccache/issues/2092
+                if e.kind() == std::io::ErrorKind::NotFound {
+                    debug!(
+                        "Error removing file from cache as it was not found: `{:?}`",
+                        remove_path
+                    );
+                } else {
+                    panic!(
+                        "Error removing file from cache: `{:?}`: {}, {:?}",
+                        remove_path,
+                        e,
+                        e.kind()
+                    )
+                }
             });
         }
         Ok(())

